frontend/src/components/projects/ProjectEditContent.js

```diff
@@ -0,0 +1,178 @@
+    action
+    .then(() => { setFormErrors({}) })
+    .catch(err => { setFormErrors(err.data || {}) })
+    .then(() => Promise.all([listTable(), summary()]))
+  }
+
+  /*
+   * Render
+   */
+
+  const title = id === undefined ?
+    'Add Project' :
+    `Update Project ${project ? project.name : id}`
+
+  const props = name =>
+    !formErrors[name] ? { name } : {
+      name,
+      hasFeedback: true,
+      validateStatus: 'error',
+      help: formErrors[name],
+    }
+
+  return (
+    <>
+      <AppPageHeader
+        title={title}
+        onBack={() => history.push(`/projects/${id || 'list'}`)}
+      />
+      <PageContent>
+        <Form
+          key={project ? 'with-project' : 'without-project'}
+          labelCol={{ span: 4 }}
+          wrapperCol={{ span: 14 }}
+          layout="horizontal"
+          initialValues={formData}
+          onValuesChange={onValuesChange}
+          onFinish={onSubmit}
+        >
+          <Form.Item label="Name" {...props("name")} rules={requiredRules} >
+            <Input />
+          </Form.Item>
+          <Form.Item label="Principal Investigator" {...props("principal_investigator")} >
+            <Input />
+          </Form.Item>
+          <Form.Item label="Requestor Name" {...props("requestor_name")} >
+            <Input />
+          </Form.Item>
+          <Form.Item label="Requestor Email" {...props("requestor_email")} rules={emailRules} >
+            <Input />
+          </Form.Item>
+          <Form.Item label="Status" {...props("status")} rules={requiredRules}>
+            <Select>
+              <Option key="Ongoing" value="Ongoing" > Ongoing </Option>
+              <Option key="Completed" value="Completed" > Completed </Option>
+            </Select>
+          </Form.Item>
+          <Form.Item label="Target End Date" {...props("targeted_end_date")} >
+            <DatePicker />
+          </Form.Item>
+          <Form.Item label="Comments" {...props("comments")}>
+            <TextArea />
+          </Form.Item>
+          {formErrors?.non_field_errors &&
+            <Alert
+              showIcon
+              type="error"
+              style={{ marginBottom: '1em' }}
+              message="Validation error(s)"
+              description={
+                <ul>
+                  {
+                    formErrors.non_field_errors.map(e =>
+                      <li key={e}>{e}</li>
+                    )
+                  }
+                </ul>
+              }
+            />
+          }
+          <Form.Item>
+            <Button type="primary" htmlType="submit">
+              Submit
+            </Button>
+          </Form.Item>
+        </Form>
+      </PageContent>
+    </>
+  );
+}
+
+function deserialize(values) {
+  if (!values)
+    return undefined
+  const newValues = {...values}
+
+  if (newValues.creation_date)
+    newValues.creation_date = moment(newValues.creation_date, 'YYYY-MM-DD')
```

Comment:
You are unlikely to ever get creation_date here ... Maybe this can apply to targeted_end_date instead ...